apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -143,3 +147,17 @@ func validateBMHName(bmhname string) error {
 
 	return nil
 }
+
+func validateDNSName(hostaddress string) error {
+
+	if hostaddress == "" {
+		return nil
+	}
+
+	_, err := bmc.GetParsedURL(hostaddress)
+	if err != nil {
+		return fmt.Errorf("host DNS name is invalid")
```

Comment:
Could you include the original error using `errors.Wrap`?

Reply:
@dtantsur I've wrapped the errors, and added a few more test cases, can you please take another look.